pkg/cluster/cluster_test.go

```diff
@@ -0,0 +1,364 @@
+
+		gomock.InOrder(
+			mockKubeClients.
+				EXPECT().
+				HasResource(machinev1.SchemeGroupVersion.WithResource("machineconfigs")).
+				Return(true, nil),
+			mockKubeClients.
+				EXPECT().
+				Get(context.TODO(), nsn, cm).
+				Do(func(_ context.Context, _ types.NamespacedName, cm *unstructured.Unstructured) {
+					err := unstructured.SetNestedStringMap(cm.Object, make(map[string]string), "data")
+					Expect(err).NotTo(HaveOccurred())
+				}),
+		)
+
+		_, err := cluster.NewCluster(mockKubeClients).OSImageURL()
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("should return the ConfigMap's osImageURL field if the machine-config-osimageurl ConfigMap can be found", func() {
+		const osImageURLValue = "value"
+
+		cm := &unstructured.Unstructured{}
+		cm.SetAPIVersion("v1")
+		cm.SetKind("ConfigMap")
+
+		gomock.InOrder(
+			mockKubeClients.
+				EXPECT().
+				HasResource(machinev1.SchemeGroupVersion.WithResource("machineconfigs")).
+				Return(true, nil),
+			mockKubeClients.
+				EXPECT().
+				Get(context.TODO(), nsn, cm).
+				Do(func(_ context.Context, _ types.NamespacedName, cm *unstructured.Unstructured) {
+					err := unstructured.SetNestedStringMap(cm.Object, map[string]string{"osImageURL": osImageURLValue}, "data")
+					Expect(err).NotTo(HaveOccurred())
+				}),
+		)
+
+		s, err := cluster.NewCluster(mockKubeClients).OSImageURL()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(s).To(Equal(s))
+	})
+})
+
+var _ = Describe("cluster_OperatingSystem", func() {
+	AfterEach(func() {
+		cache.Node.List.Items = nil
+	})
+
+	It("should return an error when feature.node.kubernetes.io/system-os_release.ID is empty", func() {
+		cache.Node.List.Items = make([]unstructured.Unstructured, 1)
+
+		_, _, _, err := cluster.NewCluster(nil).OperatingSystem()
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("should return an error when feature.node.kubernetes.io/system-os_release.VERSION_ID.major is empty", func() {
+		cache.Node.List.Items = []unstructured.Unstructured{
+			func() unstructured.Unstructured {
+				uo := unstructured.Unstructured{}
+				uo.SetLabels(map[string]string{"feature.node.kubernetes.io/system-os_release.ID": "abcd"})
+
+				return uo
+			}(),
+		}
+		_, _, _, err := cluster.NewCluster(nil).OperatingSystem()
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("should use RHEL version when it has 3 digits", func() {
+		cache.Node.List.Items = []unstructured.Unstructured{
+			func() unstructured.Unstructured {
+				uo := unstructured.Unstructured{}
+				uo.SetLabels(map[string]string{
+					"feature.node.kubernetes.io/system-os_release.ID":               "abc",
+					"feature.node.kubernetes.io/system-os_release.VERSION_ID.major": "def",
+					"feature.node.kubernetes.io/system-os_release.RHEL_VERSION":     "123",
+				})
+
+				return uo
+			}(),
+		}
+
+		o0, o1, o2, err := cluster.NewCluster(nil).OperatingSystem()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(o0).To(Equal("rhel1"))
+		Expect(o1).To(Equal("rhel123"))
```

Comment:
While this test is correct, not sure about the function's logic now. rhel version is different than appending major and minor.